lib/node/src/transport/WebSocketServer.js

```diff
@@ -31,109 +31,80 @@
-            });
+        const client = this._transportConfig.getClient();
+        const connectionLostTimeout = this._transportConfig.getConnectionLostTimeout();
 
-            // Event listener for errors
-            connection.on('error', (errorEvent) => {
-                console.log('server received error event');
-                if (this._transportCallback !== null) {
-                    this._transportCallback.onError();
-                }
-            });
+        // Event listener for an incoming message
+        client.on('message', (message) => {
+            this._handleMessage(message);
+        });
 
-            // Pong heartbeat listener
-            connection.on('pong', () => {
-                console.log('server received pong event');
-                connection.isAlive = true;
-            });
+        // Event listener for a closed connection
+        client.on('close', () => {
+            console.log('server received close event');
+            if (this._transportCallback !== null) {
+                this._transportCallback.onConnectionTerminated();
+            }
+        });
 
-            // Trigger event for connection established
-            // FIXME: pass a client context to the transport listener?
+        // Event listener for errors
+        client.on('error', (errorEvent) => {
+            console.log('server received error event');
             if (this._transportCallback !== null) {
-                this._transportCallback.onConnectionEstablished();
+                this._transportCallback.onError();
             }
-            connection.isAlive = true;
         });
 
+        // Pong heartbeat listener
+        client.on('pong', () => {
+            console.log('server received pong event');
+            client.isAlive = true;
+        });
+
+        // Trigger event for connection established
+        if (this._transportCallback !== null) {
+            this._transportCallback.onConnectionEstablished();
+        }
+
         // Detect broken connections
-        if (this._transportConfig.getConnectionLostTimeout() > 0) {
-            setInterval(() => {
-                this._ws.clients.forEach(function each (client) {
-                    if (client.isAlive === false) {
-                        console.log('server deemed client dead');
-                        return client.terminate();
-                    }
-
-                    // Assume the client is dead and ask it if it's alive
-                    client.isAlive = false;
-                    console.log('server sending ping to client');
-                    client.ping(function () {});
-                });
-            }, this._transportConfig.getConnectionLostTimeout());
+        if (connectionLostTimeout > 0) {
+            this._connectionLostInterval = setInterval(() => {
+                if (client.isAlive === false) {
+                    console.log('server deemed client dead');
+                    return client.terminate();
```

Comment:
I think this works for terminating the client but the setInterval will continuing running. I think making a call to clearInterval is also needed here. I am receiving a lot of "server deemed client dead". I shouldn't be receiving this after the client connection has been terminated.

Reply:
@russjohnson09 Thanks! I moved the logic to terminate the interval from the stop() method to the connection closed event listener.